Android/app/src/main/java/app/intra/net/socks/GoVpnAdapter.java

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2019 Jigsaw Operations LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app.intra.net.socks;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.os.Build;
+import android.os.ParcelFileDescriptor;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import app.intra.net.VpnAdapter;
+import app.intra.net.socks.TLSProbe.Result;
+import app.intra.sys.IntraVpnService;
+import app.intra.sys.LogWrapper;
+import java.io.IOException;
+import java.util.Locale;
+import tun2socks.IntraTunnel;
+import tun2socks.Tun2socks;
+
+/**
+ * This is a VpnAdapter that captures all traffic and routes it through a go-tun2socks instance with
+ * custom logic for Intra.
+ */
+public class GoVpnAdapter extends VpnAdapter {
+  private static final String LOG_TAG = "SocksVpnAdapter";
```

Comment:
Maybe rename to `"GoVpnAdapter"` to distinguish it.

Reply:
Thanks, you're right.  Done.